scripts/build.rs

```diff
@@ -11,12 +12,26 @@ fn main() {
         logging_enabled = "on";
     }
 
+    let target = env::var("TARGET").unwrap();
+
     // Builds the native MsQuic and installs it into $OUT_DIR.
-    let dst = Config::new(".")
-                 .define("QUIC_ENABLE_LOGGING", logging_enabled)
-                 .define("QUIC_TLS", "openssl")
-                 .define("QUIC_OUTPUT_DIR", "../lib")
-                 .build();
+    let mut config = Config::new(".");
+    config
+        .define("QUIC_ENABLE_LOGGING", logging_enabled)
+        .define("QUIC_TLS", "openssl")
+        .define("QUIC_OUTPUT_DIR", "../lib");
+
+    match target.as_str() {
+        "x86_64-apple-darwin" => config
+            .define("CMAKE_OSX_ARCHITECTURES", "x86_64")
+            .define("CMAKE_OSX_DEPLOYMENT_TARGET", "10.15"),
+        "aarch64-apple-darwin" => config
+            .define("CMAKE_OSX_ARCHITECTURES", "arm64")
+            .define("CMAKE_OSX_DEPLOYMENT_TARGET", "11.0"),
+        _ => &mut config
+    };
```

Comment:
Is this for macOS support?

Reply:
Yes, this allows to run eg. 'cargo build --target x86_64-apple-darwin` and so, in near future there is a need for Android targets as well.

Reply:
Ok. Thanks. Perhaps it's work updating the Cargo GitHub action to test macOS to validate?

Reply:
In cargo.yml try adding `macos-latest` to `os: [ubuntu-latest, windows-latest]`?

Reply:
Done.